backends/core/models.py

```diff
@@ -165,15 +154,15 @@ def start(self):
 
   def _cancel_all_tasks(self):
     for job in self.jobs:
-      job._cancel_tasks()
+      job.cancel_tasks()
 
   def stop(self):
     if self.status != Pipeline.STATUS.RUNNING:
       return False
     for job in self.jobs:
       job.stop()
     for job in self.jobs:
-      if job.get_status() not in [Job.STATUS.FAILED, Job.STATUS.SUCCEEDED]:
+      if job.status not in [Job.STATUS.FAILED, Job.STATUS.SUCCEEDED]:
```

Comment:
Should we make sure to fetch jobs list from the database here? And avoid job instances in local cache?

Reply:
I think in this case we can rely on "cached" versions of the job instances: the `stop()` method is invoked immediately after the corresponding HTTP request processing is started, and the job instances are just fetched from the database and fresh enough.

I propose to refactor the entire stopping logic in a separate pull request.